src/Appwrite/Platform/Workers/Deletes.php

```diff
@@ -450,6 +450,23 @@ private function deleteProject(Database $dbForConsole, callable $getProjectDB, c
             Query::equal('projectInternalId', [$projectInternalId])
         ], $dbForConsole);
 
+        // Delete VCS Installations, Repositories and Comments
+        $this->deleteByGroup('installations', [
+            Query::equal('projectInternalId', [$projectInternalId])
+        ], $dbForConsole, function (Document $document) use ($dbForConsole) {
+            $projectInternalId = $document->getAttribute('projectInternalId', '');
```

Comment:
why is this in a nested loop ? 

You can keep deletion from the `repositories` table a separate loop.. you only need the `projectInternalID` to identify the repositories

Reply:
Fixed